### PR TITLE
Implemented networking for searching for rides

### PR DESCRIPTION
**Pull request by tiffany-pann**
## Overview
- Setup the endpoint for searching for matches in the SearchViewController and updated the previous Ride model.
- Created a new Path model

## Changes Made
- Updated the previous Ride model to include a new Path model. The Path model now stores information about a ride's departure and arrival location, as well as their respective IDs. 
- Created a RideResponse in order to receive an array of Rides during GET requests. 
- Updated the MatchesViewController to properly update searches once users hit "Find" with the proper data return from the networking request (includes various changes to the class in order to properly pass in the most updated values in the Network Manager call. 
- **Dummy data for the main VC currently commented out because the Ride model is not currently finalized**

## Test Coverage
- Not yet tested because cannot authenticate a user yet so the server just returns a 401. Will follow up once authenticating users is properly implemented.

### CODE REVIEW

File: Scoop/Controller/Home/HomeViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Home/HomeViewController.swift:43]: Can we document why these functions are commented out? Do we need them? If so, why? If not, can we delete them?

File: Scoop/Controller/Home/HomeViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Home/HomeViewController.swift]: Same here. If we ever comment code we should either 1. Just delete it instead or 2. if we plan on using it in the future document that with a comment and a "// TODO:"

File: Scoop/Controller/MatchesViewController.swift
Comment (ReadeP) [inline Scoop/Controller/MatchesViewController.swift]: If we don't use any delegate functions can we just delete this extension and delete `tableView.delegate = self`

File: Scoop/Controller/MatchesViewController.swift
Comment (ReadeP) [inline Scoop/Controller/MatchesViewController.swift:53]: newline on 59

File: Scoop/Controller/Ride Posting/PostRideLocationViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Ride Posting/PostRideLocationViewController.swift:26]: Do we have to force unwrap here? This looks dangerous

File: Scoop/Controller/Ride Posting/PostRideSummaryViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Ride Posting/PostRideSummaryViewController.swift:101]: Are usernames now supported during Onboarding?

File: Scoop/Controller/SearchViewController.swift
Comment (ReadeP) [inline Scoop/Controller/SearchViewController.swift]: You can just write `.valueChanged`

File: Scoop/Controller/SearchViewController.swift
Comment (ReadeP) [inline Scoop/Controller/SearchViewController.swift]: We don't need to use `self` here

File: Scoop/Controller/SearchViewController.swift
Comment (ReadeP) [inline Scoop/Controller/SearchViewController.swift]: Change `RideResponse` to just `response` with a lowercase r

File: Scoop/Controller/SearchViewController.swift
Comment (ReadeP) [inline Scoop/Controller/SearchViewController.swift]: Can we remove `self` here for both tripDate and datePicker

File: Scoop/Controller/SearchViewController.swift
Comment (ReadeP) [inline Scoop/Controller/SearchViewController.swift]: mark this `private`

File: Scoop/Controller/SearchViewController.swift
Comment (ReadeP) [inline Scoop/Controller/SearchViewController.swift]: mark this `private`

File: Scoop/Controller/SearchViewController.swift
Comment (ReadeP) [inline Scoop/Controller/SearchViewController.swift]: mark this `private`

File: Scoop/Model/Ride.swift
Comment (ReadeP) [inline Scoop/Model/Ride.swift]: delete these two blank lines

File: Scoop/NetworkManager.swift
Comment (ReadeP) [inline Scoop/NetworkManager.swift]: This should not be `static`. With the singleton model we create a one static instance of the `NetworkManager` but still use instance functions. You shouldn't ever have to make a function static in NetworkManager

File: Scoop/NetworkManager.swift
Comment (ReadeP) [inline Scoop/NetworkManager.swift:49]: delete newline

**Review by ReadeP — CHANGES_REQUESTED** [review]
Looking good Tiffany! Please address my review comments and re-request me. Thanks!

File: Scoop/Controller/Ride Posting/PostRideSummaryViewController.swift
Comment (rs929) [reply Scoop/Controller/Ride Posting/PostRideSummaryViewController.swift:101]: Spoke with Kate earlier today, the username param in backend just takes a Cornell email now

File: Scoop/Model/Ride.swift
Comment (rs929) [inline Scoop/Model/Ride.swift]: I believe id is an Int now

File: Scoop/NetworkManager.swift
Comment (rs929) [inline Scoop/NetworkManager.swift]: Maybe use the hostEndpoint variable rather than the whole URL, since the variable is connected to the Keys.
Not sure if we should have the URL in the code straight up

**Review by rs929 — CHANGES_REQUESTED** [review]
Great Work! Just some minor changes

File: Scoop/Controller/Home/HomeViewController.swift
Comment (tiffany-pann) [reply Scoop/Controller/Home/HomeViewController.swift:43]: I uncommented them after I updated the ride model and created a getAllRides request in the Network Manager. Will be sure to document that next time.

File: Scoop/Controller/Home/HomeViewController.swift
Comment (tiffany-pann) [reply Scoop/Controller/Home/HomeViewController.swift]: Did not realize I had to document directly in the code, I left a comment about this in the Changes Made section but I will be sure to also document this as well next time around.

File: Scoop/Controller/Ride Posting/PostRideLocationViewController.swift
Comment (tiffany-pann) [reply Scoop/Controller/Ride Posting/PostRideLocationViewController.swift:26]: Should have documented this, but I had force unwrapped it because the Ride model was giving me issues but will fix this in my next PR

File: Scoop/Controller/SearchViewController.swift
Comment (tiffany-pann) [reply Scoop/Controller/SearchViewController.swift]: Is this the same thing for other .addTarget setups in this view? I saw:

`arrivalTextField.addTarget(self, action: #selector(arrivalSearch), for: UIControl.Event.touchDown)`

Should I also change that to just `.touchDown` as well?

File: Scoop/NetworkManager.swift
Comment (tiffany-pann) [reply Scoop/NetworkManager.swift]: Got it. I remembered seeing some of Pear's network functions marked as `static`. Is there a specific reason why in its case?

File: Scoop/NetworkManager.swift
Comment (tiffany-pann) [reply Scoop/NetworkManager.swift]: Yup going to fix that, thanks for pointing that out! When networking was broken at first, I had thought maybe the URL was the reason why hence ^

File: Scoop/NetworkManager.swift
Comment (tiffany-pann) [inline Scoop/NetworkManager.swift:73]: Realized just now that my indenting is off for some of these networking functions, will fix that.

File: Scoop/Controller/MatchesViewController.swift
Comment (ReadeP) [inline Scoop/Controller/MatchesViewController.swift:36]: this should be private

File: Scoop/Controller/Ride Posting/PostRideLocationViewController.swift
Comment (ReadeP) [reply Scoop/Controller/Ride Posting/PostRideLocationViewController.swift:26]: Can you add that documentation in a "// TODO:"

**Review by ReadeP — APPROVED** [review]
Looks good! Just left 2 more comments and then you should be good to merge after richie approves

File: Scoop/NetworkManager.swift
Comment (rs929) [inline Scoop/NetworkManager.swift:121]: delete new line

**Review by rs929 — CHANGES_REQUESTED** [review]
One small fix

**Review by rs929 — APPROVED** [review]

**Pull request merged**
